pkg/controller/persistentvolume/persistentvolume_controller.go

```diff
@@ -0,0 +1,113 @@
+package persistentvolume
+
+import (
+	"strings"
+
+	"github.com/openshift/ocs-operator/pkg/controller/util"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	csiExpansionSecretName      = "csi.storage.k8s.io/controller-expand-secret-name"
+	csiExpansionSecretNamespace = "csi.storage.k8s.io/controller-expand-secret-namespace"
+)
+
+var log = logf.Log.WithName("controller_persistentvolume")
+
+// Add creates a new PersistentVolume Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	r := &ReconcilePersistentVolume{
+		client:    mgr.GetClient(),
+		scheme:    mgr.GetScheme(),
+		reqLogger: log,
+	}
+
+	return r
+}
+
+// reconcilePV
+func reconcilePV(obj runtime.Object) bool {
+	pv, ok := obj.(*corev1.PersistentVolume)
+	if !ok {
+		return false
+	}
+
+	if pv.Spec.CSI != nil && strings.Contains(pv.Spec.CSI.Driver, "csi.ceph.com") {
+		secretRef := pv.Spec.CSI.ControllerExpandSecretRef
+		if secretRef == nil || secretRef.Name == "" {
+			if pv.Spec.StorageClassName != "" {
```

Comment:
Please move SC check to above, we dont want to come down or validate secretRef or its entry if SC is nil

Reply:
Sure, makes sense.

Reply:
Addressed.